app/src/sharedTest/java/org/oppia/android/app/testing/TickIconTest.kt

```diff
@@ -0,0 +1,274 @@
+import org.oppia.android.app.recyclerview.RecyclerViewMatcher.Companion.atPositionOnView
+import org.oppia.android.app.shim.ViewBindingShimModule
+import org.oppia.android.app.topic.questionplayer.QuestionPlayerActivity
+import org.oppia.android.domain.classify.InteractionsModule
+import org.oppia.android.domain.classify.rules.continueinteraction.ContinueModule
+import org.oppia.android.domain.classify.rules.dragAndDropSortInput.DragDropSortInputModule
+import org.oppia.android.domain.classify.rules.fractioninput.FractionInputModule
+import org.oppia.android.domain.classify.rules.imageClickInput.ImageClickInputModule
+import org.oppia.android.domain.classify.rules.itemselectioninput.ItemSelectionInputModule
+import org.oppia.android.domain.classify.rules.multiplechoiceinput.MultipleChoiceInputModule
+import org.oppia.android.domain.classify.rules.numberwithunits.NumberWithUnitsRuleModule
+import org.oppia.android.domain.classify.rules.numericinput.NumericInputRuleModule
+import org.oppia.android.domain.classify.rules.ratioinput.RatioInputModule
+import org.oppia.android.domain.classify.rules.textinput.TextInputRuleModule
+import org.oppia.android.domain.onboarding.ExpirationMetaDataRetrieverModule
+import org.oppia.android.domain.oppialogger.LogStorageModule
+import org.oppia.android.domain.oppialogger.loguploader.LogUploadWorkerModule
+import org.oppia.android.domain.oppialogger.loguploader.WorkManagerConfigurationModule
+import org.oppia.android.domain.question.QuestionCountPerTrainingSession
+import org.oppia.android.domain.question.QuestionTrainingSeed
+import org.oppia.android.domain.topic.FRACTIONS_SKILL_ID_0
+import org.oppia.android.domain.topic.PrimeTopicAssetsControllerModule
+import org.oppia.android.testing.CoroutineExecutorService
+import org.oppia.android.testing.TestAccessibilityModule
+import org.oppia.android.testing.TestCoroutineDispatchers
+import org.oppia.android.testing.TestDispatcherModule
+import org.oppia.android.testing.TestLogReportingModule
+import org.oppia.android.testing.profile.ProfileTestHelper
+import org.oppia.android.util.caching.testing.CachingTestModule
+import org.oppia.android.util.gcsresource.GcsResourceModule
+import org.oppia.android.util.logging.LoggerModule
+import org.oppia.android.util.logging.firebase.FirebaseLogUploaderModule
+import org.oppia.android.util.parser.GlideImageLoaderModule
+import org.oppia.android.util.parser.HtmlParserEntityTypeModule
+import org.oppia.android.util.parser.ImageParsingModule
+import org.oppia.android.util.threading.BackgroundDispatcher
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import javax.inject.Inject
+import javax.inject.Singleton
+
+private val SKILL_ID_LIST = listOf(FRACTIONS_SKILL_ID_0)
+
+/** Tests for [QuestionPlayerActivity]. */
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(application = TickIconTest.TestApplication::class, qualifiers = "port-xxhdpi")
+class TickIconTest {
```

Comment:
This test should actually be part of ``QuestionPlayerActivity``, not in its own test suite.

Reply:
It is easier for me to test it in a separate file, so once, the test case is implemented correctly, I'll shift it to ```QuestionPlayerActivity```. Is it fine to do so?"

Reply:
What actually makes it easier to be in its own file? It's more confusing for reviewers (we should ideally only review code that's actually going to be checked in), so if it's for some reason more difficult to modify existing tests it'd be good to understand why. There might be some tips we can give you to make it easier.

Reply:
@BenHenning Okay, I am moving the test case in the `QuestionPlayerActivityTest`, actually as I was making changes in the test case so that it behaves correctly (i.e test case pass on robolectric and espresso with the new changes applied and fail o both without new changes) so it was taking a lot of time to run tests on espresso but I think it can be solved by running only the required test.
But I definitely face a problem when running tests on espresso, my system lags a lot when I use emulator, and I am only able to run espresso tests on an emulator device with API level 28, my physical device has API level 29 and I am unable to run espresso tests on it. It would be great if you could help with this

Reply:
Ah, that's actually why I was asking. Running a single test at a time is a good way to make things easier.

What issues do you encounter when trying to run on the physical device? Also, it may be more efficient to go off-thread with this discussion. Can you ping me on Gitter or Hangouts & we can continue the discussion there?

Reply:
Sure